test/dynamo/test_misc.py

```diff
@@ -7339,6 +7339,102 @@ def fn(x):
         self.assertEqual(list(eager), list(compiled))
         self.assertEqual(len(counters["graph_break"]), 0)
 
+    def test_itertools_infinite_repeat(self):
```

Comment:
If you want to be super rigorous about the tests, there might be a way to adapt the tests from CPython itertools test suite here. Up to you.